hooks/gen_docs.py

```diff
@@ -1,10 +1,277 @@
+    "app",
+    "from_",
+    "to",
+    "prefix",
+    "repo_config",
+    "version",
+    "resetter_values",
+    "offset_topic",
+]
+
+log = logging.getLogger("DocumentationGenerator")
+
+#####################
+# EXAMPLES          #
+#####################
+
+DANGEROUS_FILES_TO_CHANGE = {
+    PATH_DOCS_COMPONENTS_DEPENDENCIES,
+    PATH_DOCS_COMPONENTS_DEPENDENCIES_DEFAULTS,
+    PATH_DOCS_KPOPS_STRUCTURE,
+}
+if not {
+    str(file.relative_to(PATH_ROOT)) for file in DANGEROUS_FILES_TO_CHANGE
+}.isdisjoint(set(sys.argv)):
+    is_change_present = True
+    PATH_DOCS_COMPONENTS_DEPENDENCIES.unlink(missing_ok=True)
+    PATH_DOCS_COMPONENTS_DEPENDENCIES_DEFAULTS.unlink(missing_ok=True)
+    log.warning(
+        typer.style(
+            "Changes in the dependency dir detected",
+            fg=typer.colors.RED,
+        )
+    )
+else:
+    is_change_present = False
+
+COMPONENTS_DEFINITION_SECTIONS = os.listdir(PATH_DOCS_COMPONENTS / "sections")
+PIPELINE_COMPONENT_FILE_NAMES = sorted(os.listdir(PATH_DOCS_COMPONENTS / "headers"))
+PIPELINE_COMPONENT_DEFAULTS = sorted(
+    os.listdir(PATH_DOCS_RESOURCES / "pipeline-defaults/headers")
+)
+
+
+class KpopsComponent(NamedTuple):
+    # name: str
+    attrs: list[str]
+    specific_attrs: list[str]
+
+
+def filter_sections(
+    component_name: str, sections: list[str], include_inherited: bool = False
+) -> list[str]:
+    """Find all sections that are specific to a component from a list
+
+    :param component_name: Component name
+    :param sections: Available section files, names with extension, no path
+    :param include_inherited: Whether to include sections that can be defined in
+        a parent component defaults
+    :returns: The applicable sections
+    """
+    component_sections: list[str] = []
+    for target_section in KPOPS_COMPONENTS_FIELDS[component_name]:
+        component = component_name
+        section = target_section + "-" + component + ".yaml"
+        if section in sections:
+            component_sections.append(section)
+        elif (
+            not include_inherited
+            and KPOPS_COMPONENTS_INHERITANCE_REF[component]
+            == PipelineComponent.get_component_type()
+        ):
+            section = target_section + ".yaml"
+            if section in sections:
+                component_sections.append(section)
+        elif include_inherited:
+            while component != PipelineComponent.get_component_type():
+                component = KPOPS_COMPONENTS_INHERITANCE_REF[component]
+                section = target_section + "-" + component + ".yaml"
+                if section in sections:
+                    component_sections.append(section)
+                    break
+                elif component == PipelineComponent.get_component_type():
+                    section = target_section + ".yaml"
+                    if section in sections:
+                        component_sections.append(section)
+                        break
```

Comment:
```suggestion
            while (
                component := KPOPS_COMPONENTS_INHERITANCE_REF[component]
            ) != PipelineComponent.get_component_type():
                section = target_section + "-" + component_name
                if section in sections:
                    component_sections.append(section)
                    break
```

this seems to yield the same result for me, even though the `elif` block is missing. I am wondering what's its purpose (it never seems to get executed?)

Reply:
worked without the `".yaml"` suffix to the section?? :0

It doesn't work for me. Note that this function is only used when `check_for_changes_in_kpops_component_structure()` notices a change.